lib/Runtime/Library/JSONParser.cpp

```diff
@@ -404,12 +404,13 @@ namespace JSON
                 }
 
                 // check  and consume the ending '}"
-                CheckCurrentToken(tkRCurly, ERRnoRcurly);
+                CheckCurrentToken(tkRCurly, JSERR_JsonNoRcurly);
                 return object;
             }
 
         default:
-            Js::JavascriptError::ThrowSyntaxError(scriptContext, ERRsyntax);
+            m_scanner.ThrowSyntaxError(JSERR_JsonSyntax);
+            return NULL;
```

Comment:
That is because `JSONScanner::ThrowSyntaxError` is not marked with `__declspec(noreturn)`.

Probably reasonable to change the definition of `JSONScanner::ThrowSyntaxError` to:
```c++
void __declspec(noreturn) ThrowSyntaxError(int wErr)
```